plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestRemoteQueryCommentLogging.java

```diff
@@ -0,0 +1,94 @@
+        DistributedQueryRunner distributedQueryRunner = createPostgreSqlQueryRunner(
+                postgreSqlServer,
+                Map.of(),
+                Map.of("remote-query.comment.format", "query executed by $USER"),
+                List.of(CUSTOMER, NATION));
+
+        postgreSqlServer.execute("ANALYZE " + CUSTOMER.getTableName());
+        postgreSqlServer.execute("ANALYZE " + NATION.getTableName());
+
+        return distributedQueryRunner;
+    }
+
+    @Test
+    public void testShouldLogContextInComment()
+    {
+        assertThat(postgreSqlServer.recordEventsForOperations(() -> getQueryRunner().execute("CREATE TABLE postgresql.tpch.log_nation_test_table AS (SELECT * FROM postgresql.tpch.nation)"))
+                .streamQueriesContaining("\"tpch\".\"tpch\".\"tmp_trino_"))
+                .allMatch(query -> query.endsWith("/*query executed by user*/"))
+                .size()
+                .isGreaterThanOrEqualTo(3);
+
+        assertThat(postgreSqlServer.recordEventsForOperations(() -> getQueryRunner().execute("SELECT * FROM postgresql.tpch.log_nation_test_table"))
+                .streamQueriesContaining("log_nation_test_table"))
+                .allMatch(query -> query.endsWith("/*query executed by user*/"))
+                .size()
+                .isEqualTo(1);
+
+        assertThat(postgreSqlServer.recordEventsForOperations(() -> getQueryRunner().execute("DELETE FROM postgresql.tpch.log_nation_test_table"))
+                .streamQueriesContaining("log_nation_test_table"))
+                .allMatch(query -> query.endsWith("/*query executed by user*/"))
+                .size()
+                .isEqualTo(1);
+
+        assertThat(postgreSqlServer.recordEventsForOperations(() -> getQueryRunner().execute("INSERT INTO postgresql.tpch.log_nation_test_table VALUES (1, 'nation', 1, 'nation')"))
+                .streamQueriesContaining("log_nation_test_table", "\"tpch\".\"tpch\".\"tmp_trino_"))
+                .allMatch(query -> query.endsWith("/*query executed by user*/"))
+                .size()
+                .isEqualTo(4);
+
+        assertThat(postgreSqlServer.recordEventsForOperations(() -> getQueryRunner().execute("DROP TABLE postgresql.tpch.log_nation_test_table"))
+                .streamQueriesContaining("log_nation_test_table"))
+                .allMatch(query -> query.endsWith("/*query executed by user*/"))
+                .size()
+                .isEqualTo(1);
+    }
+
+    @Test
+    public void testShouldLogContextInCommentForTableFunctionsQueryPassthrough()
+    {
+        assertThat(postgreSqlServer.recordEventsForOperations(() -> getQueryRunner().execute("SELECT * FROM TABLE( postgresql.system.query(query => 'SELECT name FROM tpch.nation WHERE nationkey = 0'))"))
+                .streamQueriesContaining("tpch.nation"))
+                .allMatch(query -> query.endsWith("SELECT \"name\" FROM (SELECT name FROM tpch.nation WHERE nationkey = 0) o/*query executed by user*/"))
```

Comment:
How about ends with `/*query executed by user*/` ?

Reply:
I wanted to make sure that we are actually logging context details in connection to the most important query, but it might be too opening this test to the implementation details.